lib/MusicBrainz/Server/Entity/Editor.pm

```diff
@@ -128,9 +128,17 @@ sub public_privileges {
 has 'email' => (
     is        => 'rw',
     isa       => 'Str',
-    predicate => 'has_email_address',
+    predicate => 'has_email_set',
 );
 
+sub has_email_address
```

Comment:
Has this exact change really been tested?
Where is `has_email_set` defined?
Where is `has_email_address` now used?

Reply:
`has_email_address` is used in the exact same places it was before. It just adds the extra checks to it.

`has_email_set` is defined here :) See the link about predicates in the PR description.

Reply:
The link doesn’t mention any `has_email_set`.

Reply:
`has_email_set` (same as `has_email_address` before) is just the name given to check the `predicate` for `email` which is just a default "method [that] tells you whether or not a given attribute is currently set"

Reply:
Tentative to clarify the commit message:

> Previously `has_email_address` was defined as a predicate for whether or
> not the email address was set, even if set as an empty string.
> 
> This patch renames the predicate to `has_email_set` and defines
> `has_email_address` as a subroutine that also checks whether or not the
> email is non-empty string.
> 
> For reference, see the documentation for Moose predicates at
> https://metacpan.org/dist/Moose/view/lib/Moose/Manual/Attributes.pod#Predicate-and-clearer-methods